geode-core/src/main/java/org/apache/geode/internal/cache/execute/FunctionContextImpl.java

```diff
@@ -37,20 +38,25 @@
 
   private String functionId = null;
 
+  private Cache cache = null;
+
   private ResultSender resultSender = null;
 
   private final boolean isPossDup;
 
   public FunctionContextImpl(final String functionId, final Object args,
       ResultSender resultSender) {
-    this.functionId = functionId;
-    this.args = args;
-    this.resultSender = resultSender;
-    this.isPossDup = false;
+    this(null, functionId, args, resultSender, false);
+  }
+
+  public FunctionContextImpl(final Cache cache, final String functionId, final Object args,
+      ResultSender resultSender) {
+    this(cache, functionId, args, resultSender, false);
   }
 
-  public FunctionContextImpl(final String functionId, final Object args, ResultSender resultSender,
-      boolean isPossibleDuplicate) {
+  public FunctionContextImpl(final Cache cache, final String functionId, final Object args,
+      ResultSender resultSender, boolean isPossibleDuplicate) {
```

Comment:
Sure we can't remove GemFireCacheImpl.getInstance now and we might not be able to ever... but in order for it to be possible, this change would need to be made anyways if I am not mistaken... not making these constructor changes will hinder the effort later.  Since the work is already done, was there a reason why we didn't want to make these signature changes?  Is it too cumbersome with the number of variables being passed in?